app/controllers/application_controller.rb

```diff
@@ -29,4 +35,17 @@ def accessible_hosts
   def known_host?
     accessible_hosts.include?(request.host) || accessible_hosts.include?(request.host_with_port)
   end
+
+  def not_found
+    render_404
+  end
+
+  def render_404
+    respond_to do |format|
+      format.html do
+        render file: Rails.root.join("public", "404"), layout: false, status: :not_found
```

Comment:
I'd prefer to not do this here. IMO we should just `render status: not_found, nothing: true` and then HAProxy should load the file to be shown in that case.

I'm not against merging this as it is though.

Reply:
Is HAProxy using the 404 page in the public folder?

Reply:
Not at the moment, but we could make it do so.